### PR TITLE
Setting up automated deployments

### PR DESCRIPTION
**Pull request by angelazhu2**
**features**: 
- docker file to set up the github actions and create a frontend and backend image to push to docker hub. 
- set up database secrets added to github secrets
- edited the user and password secrets to the codebase in Github settings. 
- there was an update to main, that lead to the compose file depending on .env files. This was rectified by adding database secrets to github secrets and took out the dependency in docker-compose. Now it needs to get added from command line and using a local .docker.env file (see "how to test" section) 
- edited readme

**Goal**: Set up a continuous building pipeline for the ds4cg inaturalist code base so that when there is a new version of the code it will build two docker images and push them to our docker hub. 

**how to test**: I tested by creating a branch and changing the requirements to build the images when there's a new push to a test repo. 
- test locally: by running instances on local docker app. 
-- `docker compose build` then `docker compose --env-file ./.docker.env up` with local .docker.env file with secrets. tested the UI and made sure APIs still work. 
- tested that pipeline works with new tags :https://github.com/UMassCDS/ds4cg2024-inaturalist/actions/runs/10256595317 & pr: https://github.com/UMassCDS/ds4cg2024-inaturalist/pull/65

dockerhub: https://hub.docker.com/search?q=umasscds

### CODE REVIEW

File: docker-compose.yml
Comment (ginic) [inline docker-compose.yml:4]: I know you updated this for testing purposes, but let's not merge it in until you're ready to 'cut a release'.
```suggestion
    image: umasscds/inatator_frontend:0.1.0
```

File: docker-compose.yml
Comment (ginic) [inline docker-compose.yml:14]:
```suggestion
    image: umasscds/inatator_backend:0.1.0
```

**Review by ginic — CHANGES_REQUESTED** [review]
Thanks, @angelazhu2 ! The Github workflow looks good and I saw that you tested it and pushed images to Docker Hub. I'm going to delete those v0.2.0 tagged images for now, until we do a proper release on Github.

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:25]: This shouldn't be necessary and it's also not secure. The environment variables can be set using the `--env` or `--env_file` options by the person running the container or via referencing variables in the docker compose file. See https://www.baeldung.com/ops/docker-container-environment-variables or https://www.baeldung.com/ops/docker-compose#managing-environment-variables for more details.

**Review by ginic — CHANGES_REQUESTED** [review]
Thanks, @angelazhu2 ! The `Create env file` step needs to be removed, it's not the appropriate way to configure environment variables in Docker. I don't quite understand why you thought this was needed, so it would probably be helpful to meet and discuss so we can find a better solution.

**Review by cheerstopaula — APPROVED** [review]

File: docker-compose.yml
Comment (SergePogorelov) [inline docker-compose.yml:7]: after changing the `env_file` to these ` environment` variables, do we need to run app and docker compose locally differently?
in other words, do we need to change the instructions in the README file?

**Review by SergePogorelov — CHANGES_REQUESTED** [review]
left a question about README,
and also please don't forget to update CHANGELOG file.

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:21]: There's a more modern way to do this now, the example I sent you was a little older. It's probably better to use this: 
```suggestion
      - name: Login to Docker Hub
        uses: docker/login-action@v3
        with:
          username: ${{ vars.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:14]: We can use the most recent version here:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:19]: These lines are only necessary if we want to do multi-platform builds, which I don't think we do. 
```suggestion
```

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:26]: This isn't necessary, see my next commend on `docker compose`
```suggestion
```

File: .github/workflows/create-docker-images.yml
Comment (ginic) [inline .github/workflows/create-docker-images.yml:30]: Docker compose should already be available, it's just that you don't need the hyphen 😅 See https://github.com/orgs/community/discussions/116610 for more details.
```suggestion
          docker compose -f docker-compose.yml build
          docker compose -f docker-compose.yml push
```

File: docker-compose.yml
Comment (ginic) [reply docker-compose.yml:7]: Yes, nice catch, the Readme instructions should be updated as well!

**Review by ginic — CHANGES_REQUESTED** [review]
Thanks for figuring out the environment variable stuff, @angelazhu2! I was able to test the images you pushed to Docker Hub for 0.1.0 and everything worked well. 

There are still some things that need to be addressed:
- The Github action can be simplified a lot. I've put everything in suggestions. 
- The README needs to be updated with the instructions for passing the .env file

Once that's done and this PR is merged, we should: 
- Delete the tags and releases created for testing, both on GitHub and Docker Hub. 
- Finalize the contribution guidelines and release process Google Doc
- Restore the tag for the original v0.1.0 (we'll be able to do this by creating a release/tag for v0.1.0 on the commit that updated the changelog for v0.1.0)
- Release a v1.0.0 version (probably the final version we'll deliver this summer)

File: .github/workflows/create-docker-images.yml
Comment (oz03-hub) [reply .github/workflows/create-docker-images.yml:19]: I am not familiar with the github worklow portion, but we want to target linux/amd64 platform while building, without these lines would it still work?

**Review by ginic — APPROVED** [review]
Looks good! Nice work, @angelazhu2 !

**Pull request merged**
